### DISCUSSION TITLE
Typescript Samples

### DISCUSSION THREAD

**Discussion by 0xbrayo** [firebase/functions-samples#1101] [Ideas]
I learned to write Firebase cloud functions based off the youtube firecasts videos. So it's kind of discouraging to see so much of the samples are in Javascript while Typescript is actually what's recommended according to your videos. Maybe reimplement some if not all in Typescript. It's basically minor tweaks anyway. Have both Javascript and Typescript samples.

**Comment by william-ferguson-au** [comment]
Ditto. Surely nudging developers towards using Typescript is the obvious way to go, assuming of course that you are interested in generating strong maintainable code. But when all the examples are in Javascript  it can become a real fight just to get going.
